src/libraries/System.Diagnostics.DiagnosticSource/tests/System.Diagnostics.DiagnosticSource.Tests.csproj

```diff
@@ -14,16 +14,6 @@
     <Compile Include="TestNotSupported.cs" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="..\src\System\Diagnostics\Metrics\Aggregator.cs" Link="Aggregator.cs" />
```

Comment:
Interesting! I can revert it. `InternalsVisibleTo` does seem to be used around the repo though 🤷 https://github.com/search?q=repo%3Adotnet%2Fruntime%20assembly%3A%20InternalsVisibleTo(%22&type=code

Reply:
> InternalsVisibleTo does seem to be used around the repo though

Minimally, and really only in places where we inherited it and haven't had the time to remove it.

Reply:
I just pushed some changes. I didn't want to just include the entire `DiagnosticSourceEventSource` class in the tests project because then there would be 2 `EventSource`s with the same name in the process. I could have just spun off the sampler stuff into its own file and included it but I felt this code was just in dire need of some refactoring. I tried to separate things out so there is better separation of concerns/less spaghetti. I can revert that and do less but IMO it leaves the camp site in a better state. Check it out, LMK!